aspnetcore/blazor/components/render-modes.md

```diff
@@ -0,0 +1,460 @@
+
+## Render mode propagation
+
+Render modes propagate down the component hierarchy.
+
+Rules for applying render modes:
+
+* The default render mode is Static. 
+* The interactive Server, WebAssembly, and Auto render modes can be used from a Static component, including using different render modes for sibling components. 
+* You can't switch to a different interactive render mode in a child component. For example, a Server component can't be a child of a WebAssembly component.
+* Parameters passed to an interactive child component from a Static parent must be JSON serializable. This means that you can't pass render fragments or child content from a Static parent component to an interactive child component.
+
+The following examples use a non-routable, non-page `SharedMessage` component. The render mode agnostic `SharedMessage` component doesn't apply a render mode with an [`@attribute` directive](xref:mvc/views/razor#attribute). If you're testing these scenarios with a Blazor Web App, place the following component in the app's `Components` folder.
+
+`SharedMessage.razor`:
+
+```razor
+<p>@Greeting</p>
+
+<button @onclick="UpdateMessage">Click me</button> @message
+
+<p>@ChildContent</p>
+
+@code {
+    private string message = "Not clicked yet.";
+
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    [Parameter]
+    public string Greeting { get; set; } = "Hello!";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+### Render mode inheritance
+
+If the `SharedMessage` component is placed in a statically-rendered parent component, the `SharedMessage` component is also rendered statically and isn't interactive. The button doesn't call `UpdateMessage`, and the message isn't updated.
+
+`RenderMode5.razor`:
+
+```razor
+@page "/render-mode-5"
+
+<SharedMessage />
+```
+
+If the `SharedMessage` component is placed in a component that defines the render mode, it inherits the applied render mode.
+
+In the following example, the `SharedMessage` component is interactive over a SignalR connection to the client. The button calls `UpdateMessage`, and the message is updated.
+
+`RenderMode6.razor`:
+
+```razor
+@page "/render-mode-6"
+@attribute [RenderModeServer]
+
+<SharedMessage />
+```
+
+### Child components with different render modes
+
+In the following example:
+
+* The first `SharedMessage` child component is interactive with Server rendering. The component immediately appears and is interactive when the page is displayed in the browser.
+* The second `SharedMessage` child component is interactive with WebAssembly rendering. The component is rendered and interactive *after* the Blazor app bundle is downloaded and the .NET runtime is active on the client.
```

Comment:
This bullet is a bit ambiguous about whether the component is rendered before or after the app and runtime get loaded on the client.

Both components should immediately appear because they are prerendered. The first will be interactive once the circuit is established, which is usually pretty quick. The second will be interactive after the runtime and app bundle have been downloaded and loaded.